src/overlays/actors/ovl_En_Guest/z_en_guest.c

```diff
@@ -20,23 +34,200 @@ const ActorInit En_Guest_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Destroy.s")
+static ColliderCylinderInit_Set3 colliderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 10, 60, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry initChain[] = {
+    ICHAIN_U8(unk_1F, 6, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 500, ICHAIN_STOP),
+};
+
+UNK_PTR D_80A50BA4[] = {
+    &D_060005FC,
+    &D_060006FC,
+    &D_060007FC,
+};
+
+void EnGuest_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+
+    if ((gSaveContext.infTable[7] & 0x40) != 0) {
+        Actor_Kill(&this->actor);
+    } else {
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OS_ANIME);
+        if (this->objBankIndex < 0) {
+            osSyncPrintf(VT_COL(RED, WHITE));
+            osSyncPrintf("%s[%d] : バンクが無いよ！！\n", "../z_en_guest.c", 129); // No such bank!!
+            osSyncPrintf(VT_RST);
+            __assert("0", "../z_en_guest.c", 132);
+        }
+    }
+}
+
+void EnGuest_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnGuest_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+    u32 padding;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex) != 0) {
+        this->actor.flags &= ~0x10;
+        Actor_ProcessInitChain(&this->actor, initChain);
+
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060000F0, NULL, this->limbDrawTable, this->transitionDrawTable,
+                         0x10);
+        gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objBankIndex].segment);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060042AC, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_060042AC.genericHeader), 0, 0.0f);
+
+        this->actor.draw = &EnGuest_Draw;
+        this->actor.update = &func_80A505CC;
+
+        Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+        Collider_SetCylinder_Set3(globalCtx, &this->colliderCylinder, &this->actor, &colliderInit);
+
+        Actor_SetHeight(&this->actor, 60.0f);
+
+        this->unk_30E = 0;
+        this->unk_30D = 0;
+        this->unk_2CA = 0;
+        this->actor.textId = 0x700D;
+        this->actionFunc = func_80A50518;
+    }
+}
+
+void func_80A5046C(EnGuest* this) {
+    if (this->unk_30D == 0) {
+        if (this->unk_2CA != 0) {
+            this->unk_2CA--;
+        } else {
+            this->unk_30D = 1;
+        }
+    } else {
+        if (this->unk_2CA != 0) {
+            this->unk_2CA--;
+        } else {
+            this->unk_30E += 1;
+            if (this->unk_30E >= 3) {
+                this->unk_30E = 0;
+                this->unk_30D = 0;
+                this->unk_2CA = (s32)Math_Rand_ZeroFloat(60.0f) + 20;
```

Comment:
cast needed? if 2CA is a s32 id think its implicit but could be wrong

Reply:
Yeah integer 20 needs to be added, not 20f